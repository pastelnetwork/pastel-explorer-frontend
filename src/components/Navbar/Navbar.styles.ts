import styled from 'styled-components/macro';

export const NavContainer = styled.nav`
  position: sticky;
  top: 0;
  display: flex;
  justify-content: space-between;
  width: 100%;
  padding-left: 16px;
  padding-right: 12px;
  background: ${props => props.theme.sidebar.menu.background};
  z-index: 100;
  box-shadow: 0 14px 20px rgb(16 16 16 / 6%);

  .cluster-button {
    padding: 0;
    border: 0;
    min-width: unset;

    &:hover {
      background: transparent;

      .cluster-icon {
        fill: ${props => props.theme.sidebar.menu.hover};
      }
    }
  }

  .cluster-icon {
    fill: ${props => props.theme.sidebar.menu.default};
    transition: all 0.5s ease;

    ${props => props.theme.breakpoints.down('sm')} {
      width: 22px;
    }
  }
`;

export const BrandLogo = styled.img`
  width: 120px;
`;

export const Menu = styled.div`
  display: flex;
  z-index: 3;

  ${props => props.theme.breakpoints.down('xs')} {
    align-items: flex-start;
    margin-top: 16px;
  }

  .main-menu {
    display: flex;
    align-items: center;
    height: 100%;

    & > .MuiDrawer-paper {
      position: relative;
      display: flex;
      align-items: center;
      top: unset;
      right: unset;
      left: unset;
      height: auto;
      flex-direction: unset;
      background: transparent;
      border-right: 0;
      overflow: unset;
    }

    & .MuiList-root {
      margin-top: -5px;
    }

    & .disable-padding {
      padding: 0;
    }
  }
`;
