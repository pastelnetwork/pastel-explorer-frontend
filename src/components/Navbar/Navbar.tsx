import React, { useState } from 'react';
import { NavLink } from 'react-router-dom';
import { Box, Hidden } from '@material-ui/core';
import Grid from '@material-ui/core/Grid';
import { Menu as MenuIcon } from '@material-ui/icons';

import SearchBar from '@components/SearchBar/SearchBar';
import Sidebar from '@components/Sidebar/Sidebar';
import { RouteType } from '@utils/types/routes';
import { translate } from '@utils/helpers/i18n';
import { useSelector } from 'react-redux';
import { getThemeState } from '@redux/reducers/appThemeReducer';
import * as ROUTES from '@utils/constants/routes';
import * as SearchBarStyles from '@components/SearchBar/SearchBar.styles';

import PastelLogoWhite from '@assets/images/pastel-logo-white.svg';
import PastelLogo from '@assets/images/pastel-logo.svg';

import * as Styles from './Navbar.styles';
import * as SidebarStyles from '../Sidebar/Sidebar.styles';

export default function Navbar({ routes }: { routes: Array<RouteType> }): JSX.Element {
  const [mobileOpen, setMobileOpen] = useState(false);
  const handleDrawerToggle = () => {
    setMobileOpen(!mobileOpen);
  };

  const isDarkMode = useSelector(getThemeState).darkMode;

  return (
    <Styles.NavContainer>
      <Styles.Menu>
        <SidebarStyles.Brand
          className="desktop-logo"
          component={NavLink}
          to={ROUTES.EXPLORER}
          button
        >
          <Box ml={1}>
            <SidebarStyles.BrandLogo
              src={isDarkMode ? PastelLogoWhite : PastelLogo}
              alt="Pastel Logo"
            />
          </Box>
        </SidebarStyles.Brand>
        <Hidden smUp>
          <Grid item>
            <SearchBarStyles.IconButton
              color="inherit"
              aria-label={translate('components.searchBar.openDrawer')}
              onClick={handleDrawerToggle}
            >
              <MenuIcon />
            </SearchBarStyles.IconButton>
          </Grid>
        </Hidden>
        <Hidden mdUp implementation="js">
          <Sidebar
            routes={routes}
            variant="temporary"
            open={mobileOpen}
            onClose={handleDrawerToggle}
          />
        </Hidden>
        <Hidden smDown implementation="css">
          <Sidebar routes={routes} />
        </Hidden>
      </Styles.Menu>
      <SearchBar isDarkMode={isDarkMode} />
    </Styles.NavContainer>
  );
}
