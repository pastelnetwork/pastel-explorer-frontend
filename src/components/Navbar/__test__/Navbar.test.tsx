import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';

import 'jest-styled-components';

import Navbar from '../Navbar';

const mockStore = configureMockStore();
const store = mockStore({});

function MyComponent() {
  return <div>My component</div>;
}

describe('components/Navbar', () => {
  const routes = [
    {
      id: 'my',
      path: '/my-page',
      children: null,
      component: MyComponent,
    },
  ];
  const wrapper = shallow(
    <Provider store={store}>
      <Navbar routes={routes} />
    </Provider>,
  );

  test('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
