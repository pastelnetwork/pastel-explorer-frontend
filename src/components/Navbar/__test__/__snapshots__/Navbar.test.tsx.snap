// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Navbar renders correctly 1`] = `ShallowWrapper {}`;
